pleasew

```diff
@@ -133,7 +146,7 @@ fi
 
 printf >&2 '%bDownloading Please %s to %s...%b\n' "${GREEN}" "${VERSION}" "${DIR}" "${RESET}"
 mkdir -p "${DIR}"
-curl -fsSL "${PLEASE_URL}" | tar -xJpf- --strip-components=1 -C "${DIR}"
+curl -#fSL "${PLEASE_URL}" | tar -xJpf- --strip-components=1 -C "${DIR}"
```

Comment:
Do you know what curl will do here if the terminal isn't interactive? i.e. will this force the progress bar to show always or will it still select intelligently?

Reply:
AFAIK, it forces the progress bar to show (tried with a non-interactive docker container). Let me know if that causes an issue and I should remove it.

Ultimately the script should use either `curl` or `wget` depending on the availability on the user's system. Should I add that in this PR ?

Reply:
Okay, maybe let's see how it goes. I'm mildly worried about it producing heaps of lines of output in a scenario like that but hopefully it's fine.

I like the idea of trying to detect `curl` or `wget`, don't mind if  you want to do that in this PR or another.

Reply:
> I like the idea of trying to detect `curl` or `wget`, don't mind if you want to do that in this PR or another.

Done ! Other utilities such as `grep` and `sed` should be present in POSIX systems and don't require checks.

> Okay, maybe let's see how it goes. I'm mildly worried about it producing heaps of lines of output in a scenario like that but hopefully it's fine.

This is only to let the user know that *something* is happening (in my case, slow internet connection). As a compromise, would disabling the progress bars in CI be enough ? Otherwise let me know and I'll remove them entirely.

Reply:
What about just removing the `s`? That should leave it as a default, which is maybe less obvious than a progress bar, but I think it will more or less do "the right thing" in most cases. WDYT?

Reply:
Yeah that works, it should reduce the amount of output for non-interactive terminals while visually signaling to the user that it's not stuck somewhere. I like it !

`wget` should have similar behavior by default according to its manpage where it changes the progress indicator for non-interactive terminals. Nothing to do handle on our end.